src/frontend/screens/MapScreen/GPSPermissions/GPSForegroundPermissionDisabled.tsx

```diff
@@ -26,26 +26,28 @@ const m = defineMessages({
 });
 
 interface GPSPermissionsDisabled {
-  setIsGranted: React.Dispatch<React.SetStateAction<boolean | null>>;
+  setForegroundStatusGranted: React.Dispatch<
+    React.SetStateAction<boolean | null>
+  >;
 }
-export const GPSPermissionsDisabled: React.FC<GPSPermissionsDisabled> = ({
-  setIsGranted,
-}) => {
+export const GPSForegroundPermissionDisabled: React.FC<
+  GPSPermissionsDisabled
+> = ({setForegroundStatusGranted}) => {
   const {formatMessage} = useIntl();
-  const requestForLocationPermissions = async () => {
-    const [foregroundPermission, backgroundPermission] = await Promise.all([
-      Location.requestForegroundPermissionsAsync(),
-      Location.requestBackgroundPermissionsAsync(),
-    ]);
-    if (foregroundPermission.granted && backgroundPermission.granted) {
-      setIsGranted(true);
-    } else if (
-      !foregroundPermission.canAskAgain ||
-      !backgroundPermission.canAskAgain
-    ) {
+
+  async function askForegroundLocationPermission() {
+    const foregroundPermission =
+      await Location.requestForegroundPermissionsAsync();
+
+    if (foregroundPermission.granted) {
+      setForegroundStatusGranted(true);
+      return;
+    }
+
+    if (foregroundPermission.canAskAgain) {
```

Comment:
same comment about this being flipped

```suggestion
    if (!foregroundPermission.canAskAgain) {
```

Reply:
https://github.com/digidem/comapeo-mobile/pull/810/commits/290ca88b575d19f0103839b117e82d6bb254d3c8